### PR TITLE
Backwards compatibility for vendors and orgs without certs

### PR DESCRIPTION
**Pull request by reinkrul**
Allow:

1. Vendors without an active certificate to register organizations
2. Organizations without certificates to register endpoints

### CODE REVIEW

File: pkg/admin.go [hunk 1]
```diff
@@ -81,35 +81,45 @@ func (r *Registry) VendorClaim(vendorID string, orgID string, orgName string, or
...
-	if err != nil {
-		return nil, errors2.Wrap(err, ErrJWKConstruction.Error())
+		jwkAsMap, err := certToJwkMap(certificate, cert.OrganisationCertificate)
+		orgKeys = append(orgKeys, jwkAsMap)
+		if err != nil {
+			return nil, errors2.Wrap(err, ErrJWKConstruction.Error())
+		}
+	} else {
+		// Vendor has no certificates, we just make sure the org has a plain JWK without X.509 certificate, either
+		// provided or freshly generated.
```
Comment (woutslakhorst) [inline pkg/admin.go:115]: Add a comment that this should be removed when all vendors have certificates.

File: pkg/admin.go [hunk 2]
```diff
@@ -119,7 +129,11 @@ func (r *Registry) VendorClaim(vendorID string, orgID string, orgName string, or
 		OrgKeys:          orgKeys,
 		Start:            time.Now(),
 	}, func(dataToBeSigned []byte, instant time.Time) ([]byte, error) {
-		return r.signAsOrganization(orgID, orgName, dataToBeSigned, instant)
+		if len(certificates) > 0 {
```
Comment (woutslakhorst) [inline pkg/admin.go:132]: could this check be in `signAsOrganization`? That would remove the double returns here.

Comment (reinkrul) [reply pkg/admin.go:132]: fixed!

File: pkg/admin.go (hunk 1 above)
Comment (reinkrul) [reply pkg/admin.go]: Fixed, specified in the same words as in the comment as below (to ease searching for what to remove).

File: pkg/admin.go [hunk 3]
```diff
@@ -81,35 +81,47 @@ func (r *Registry) VendorClaim(vendorID string, orgID string, orgName string, or
...
 	}
-	certificates := vendor.GetActiveCertificates()
-	if len(certificates) == 0 {
-		return nil, fmt.Errorf("vendor has no active certificates (id = %s)", vendorID)
-	}
 
-	// If no keys are supplied, make there's a key in the crypto module for the organisation
-	if orgKeys == nil || len(orgKeys) == 0 {
-		logrus.Infof("No keys specified for organisation (id = %s). Keys will be generated or loaded from crypto module.", orgID)
+	// If no keys are supplied, make sure there's a key in the crypto module for the organisation
```
Comment (stevenvegt) [inline pkg/admin.go:85]: What if the certificates are expired? Should we generate new keys? This feels like a explicit administration step, not a side effect of making a vendorClaim

Comment (reinkrul) [reply pkg/admin.go:85]: When certificates are expired but the keys haven't (according to the key rotation scheme in place), certificates should be reissued. When keys are expired by the key rotation scheme, new keys must be generated and new certificates issued.

That is not part of this bugfix however and we don't support it yet. We need to be able to update the registry for that.

**Comment by reinkrul** [comment]
> also create an issue for removing the backwards compatibility when all parties have migrated?

I'll create an issue and refer to it from the comments in code.

**Pull request merged**